src/main/java/htsjdk/samtools/cram/structure/CramCompressionRecord.java

```diff
@@ -152,13 +152,22 @@ public String toString() {
         return stringBuilder.toString();
     }
 
-    public int getAlignmentSpan() {
-        if (alignmentSpan < 0) calculateAlignmentBoundaries();
+    public int getAlignmentSpan(final boolean APDelta) {
+        if (alignmentSpan < 0) {
+            calculateAlignmentBoundaries(APDelta);
+        }
         return alignmentSpan;
     }
 
-    void calculateAlignmentBoundaries() {
-        if (isSegmentUnmapped()) {
+    public int getAlignmentEnd(final boolean APDelta) {
+        if (alignmentEnd < 0) {
```

Comment:
What are the possible interpretations of `alignmentEnd < 0` ? Is it possible to replace tests like this with something stricter that checks for a specific value ? Could we always initialize `alignmentEnd` to a specific value indicating that not yet known due to APDelta weirdness ? Either way this could use some beefed up comments describing what its doing.

Reply:
This is a check to see if these values have been initialized.  I'll make that more explicit.

More generally, this is a part of the horrible mess of `CramCompressionRecord` alignment boundary calculation which I hope to get to during the CramCompressionRecord refactoring.